### DISCUSSION TITLE
What's the exact default preset for livestream?

### DISCUSSION THREAD

**Discussion by sanj3k** [gopro/labs#824] [General]
Hi, as above mentioned I'd really love to know more than only the resolution and that Hypersmooth is on. 

What about the framerate is it 30 or 60? Is it the same in all three resolutions?

Can some settings still come through I was just testing livestream locally and can guarantee it's either wide gamut what and or even flat colors too. I mean for regular recordings it's nice but I'd have assumed in the livestream it would look more contrasty and saturated. Is this a bug or even intended and if yes then which other settings are getting passed through into the livestream

**Comment by dnewman-gpsw** [comment]
Labs doesn't modify Livestream default frame-rates. I don't know, likely 30.

What Labs commands are you trying to run with Livestream?

**Reply by sanj3k** [reply]
I started the livestream with the quik app but with a Labs enabled gopro running wide=1, nr01=1. 

There is this feature to save the the footage to sdcard. Can I then run it through metadata extractor? Because I'd assume that these flags should have disappeared but I'm 100% sure after what i have seen that some of them didn't have been overwritten

**Reply by dnewman-gpsw** [reply]
Remember wide and nr01, are not commands, only !MWIDE=1 and !MNR01=1 are, and only for HERO11 and 12. They will be displayed on boot and can be display with oMFEAT=4 with the latest Labs cameras.

**Reply by sanj3k** [reply]
Understand but this doesn't answer my question. First of all what exactly is overwritten by livestream default? What i know is resolution and stabilization for example. 

And since there is this feature save what you record in the livestream. Could i use this to then extra metadata to check for frame rate or which other labs flags were enabled? As what i have tried to describe i would have assumed together with the default livestream profile would have been my image overwritten say with GoPro colors but it wasn't the case i could 100% see it looked like usually when i run WIDE gamut, flat colors, LOGB 400 and NR01

**Reply by dnewman-gpsw** [reply]
I'm not sure exactly what your question is.  If you see the image you should know what is active, seeing the metadata with a local capture is something you can do, but the metadata will not guarantee that feature was applied.  WYSIWYG, as livestream is its own thing. So the only way to know for sure is to look at the image. NR01 is likely working, although not ideal for livestream as a noise increase will impact the bit-rate requirements (extremely limited in LS.) WIDE will likely work, although WIDE gamut on 8-bit is also not ideal. LOGB with Livestream, may work, as the curve should be applied, however as you can't set Flat, tonemapping is likely active, you may have to mess with TONE.  I guess the better question is why are you going this?

**Reply by sanj3k** [reply]
The mentioned parameters are usually set up for other shots i take. 

Now everyone says livestream is a totally different thing which is basically not configurable like you would set stuff up before but it will be overwritten and thats not happening the above mentioned flags are still valid means for going live I'd have to set the camera to totally different preset because it can't overwrite the existing settings. 

As i have several times repeated myself it's not my intention to flat logb nr 1% into a livestream. I was expecting more of the overwriting behavior than actually takes place. It's just not consistent or can I within livestream set up with qr codes the shutter, iso or white balance values?

**Reply by dnewman-gpsw** [reply]
Sorry, I don't fully understand. I think you are now saying you want Livestream not to honor Labs settings like LOGB and WIDE? 

"Now everyone says livestream is a totally different thing" - who is everyone? 

"basically not configurable" - Livestream doesn't have a preset to configure, which is why there are no Protune controls for it. 

 All Labs settings in the form !Mwxyz= are global, they don't use presets, they should impact livestream, but that has not been tested.

**Reply by sanj3k** [reply]
Oh wow that totally makes sense then that the labs settings are applied globally which is why I have seen their effect even though the livestream started. That's interesting! I will experiment a bit maybe you have some time for it too. 

And yes I would have loved that only during livestream these global options wouldn't have been set because it doesn't make sense at all and when livestream is over they'd be present again. But with your explanation it totally makes sense

**Reply by sanj3k** [reply]
And regarding this topic just have a clear statement I have now run GoPro metadata extractor on yesterday's file which gets saved through the save livestream to SD card option plus i have recorded two modes in which i usually record offline. 

Is this a viable measure to compare the consequences of switching livestream mode on? Because if you run these extracted metadata through a diff you can clearly see expected changes in the protune part like auto WB, shutter, turning on Hypersmooth. What's interesting this file has been saved as a 1080/50 does this then also mean the livestream was in high frame rate? 

Livestream safety to recorded to the sdcard file metadata https://pastebin.com/XsbAFWH1

Complete regular approach of recording in 5k25 and 4k50 
https://pastebin.com/NimTJftE
https://pastebin.com/ZxuuRksj

Only differences besides protune related flags the fov sensor read out time etc is different but i think this makes sense. Only flag i didn't understand was PRNA 10 -> 18, DZMX 1 -> 1.4, in 5k mode EMPT 2 something payload related, USRD.

**Reply by dnewman-gpsw** [reply]
Livestream records 50/60, but streams 25/30fps.   
PRNA - is a preset number, Livestream has an internal preset with fixed properties.  
DZMX - Digital Zoom MaX , the amount of allowed zoom
EMPT - placeholder
USRD - USeR Data

**Comment by sanj3k** [comment]
Are microphone settings also part of protune or available as global commands? 

I don't have a mediamod but thought of getting one to use external microphone within a livestream. I have found a reddit thread where people have observed a switchback to default internal microphones within the Livestreaming mode. That would defeat the whole purpose of the mediamod for me.

Could this really be a thing? Do you have a mediamod to verify and if yes could this maybe still worked around with labs global commands then?

**Reply by dnewman-gpsw** [reply]
This is not a Labs question as there is nothing in Labs that would change the default behavior of microphone selection.

**Reply by sanj3k** [reply]
Perfect! It's just this kind of information aren't bundled at one place and I'm lucky i have you. Thanks a lot! Also for answering the other questions. GoPro owe you a lot
